ext/mysqli/mysqli.c

```diff
@@ -1200,16 +1200,9 @@ void php_mysqli_fetch_into_hash(INTERNAL_FUNCTION_PARAMETERS, int override_flags
 			fci.param_count = 0;
 			fci.named_params = NULL;
 
-			if (ctor_params && Z_TYPE_P(ctor_params) != IS_NULL) {
+			if (ctor_params && zend_hash_num_elements(Z_ARRVAL_P(ctor_params)) != 0) {
```

Comment:
This check should be at the `ctor_params` below. (Checking it here is okay, but doesn't make a functional difference, so I would leave it off).